File: server/java-jaxrs/src/main/java/com/wordnik/api/EventsApiService.java
package com.wordnik.api;

import com.wordnik.model.*;

public class EventsApiService {
  public static Event addEvent(Event body) {
    return body;
  }

  public static Event findEventById(Long eventId) {
    Event event = new Event();

    event.setId(123);
    event.setName("Sample event");
    event.setStartDate(new java.util.Date());
    event.setMaxAttendees(100);

    return event;
  }

  public static ApiResponseMessage deleteEvent(Long eventId) {
    return new ApiResponseMessage(ApiResponseMessage.OK, "event deleted");
  }

  public static EventStatus getEventStatus(Long eventId) {
    EventStatus status = new EventStatus();

    status.setYesCount(100);
    status.setMaybeCount(25);
    status.setNoCount(3);
    status.setInvitedCount(150);

    return status;
  }
}

File: server/scala-scalatra/src/main/scala/apis/EventsApiService.scala
package apis

import models._

import java.util.Date
import scala.collection.mutable.HashMap

object EventsApiService {
  val events = HashMap[Long, Event](
    1L -> Event(Some(1), Some("a friendly event"), Some(new Date()), Some(100)),
    2L -> Event(Some(2), Some("an unfriendly event"), Some(new Date()), Some(25)),
    3L -> Event(Some(3), Some("a mystery event"), Some(new Date()), Some(75))
  )

  def addEvent(event: Event) = {
    if(events.contains(event.id.get))
      ApiResponse(400, "an event with that ID already exists!")
    else {
      events += event.id.get.toLong -> event
      event
    }
  }

  def findEventById(eventId: Long) = {
    events.getOrElse(eventId, ApiResponse(400, "that event doesn't exist!"))      
  }

  def deleteEvent(eventId: Long) = {
    events.remove(eventId)
  }

  def getEventStatus(eventId: Long) = {
    events.contains(eventId) match {
      case true => EventStatus(Some(1), Some(23), Some(13), Some(98))
      case false => EventStatus(None, None, None, None)
    }
  }
}
